Added EarthDesk.prefPane v7.1

amatos opened the pull request:
### Checklist

- [x] The commit message includes the cask’s name and version.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.

Additionally, when **adding a new cask**:

- [x] Checked there are no open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there are no closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,11 @@
+  url 'http://mirror.xericdesign.com/software/earthdesk.dmg'
+  name 'EarthDesk'
+  homepage 'http://www.xericdesign.com/earthdesk.php'
+  license :commercial # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
```
Remove the comment (as it says).

vitorgalvao commented:
It’s missing `uninstall`. There are [scripts](https://github.com/caskroom/homebrew-cask/tree/master/developer/bin) and [documentation](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md) geared for that.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,11 @@
+cask 'earthdeskprefpane' do
+  version '7.1'
```
Seeing as `url` doesn’t point to a specific version, this cask should use `version :latest` and `sha256 :no_check`. [See an example](https://github.com/caskroom/homebrew-cask/blob/f25b6babcd398abf48e33af3d887b2d00de1d661/Casks/kibako.rb).

vitorgalvao replied:
Make this `version '7.1'`.

amatos commented:
I've changed the comment, version and sha256 entries as suggested.  I'm not sure how to do an uninstaller.  The app installs the prefpane, but since it's not a standard .pkg, I'm not sure how to handle an uninstaller.  Do you have any suggestions or advice?

Thanks!

vitorgalvao commented:
> I'll hold off on pushing a commit back up to github until I have that resolved.

You should always be pushing, in case there’s something else that needs fixing, and then it can be done in parallel. There’s no advantage to holding off.

> I'm not sure how to handle an uninstaller. Do you have any suggestions or advice?

Unfortunately, being a `.app` installer makes things difficult. You’ll have to find complete uninstall instructions on the website, forum, or asking the developer directly and remaking them manually. Without an `uninstall` the cask will not be merged, though.

reitermarkus commented:
Does it install anything besides the `.prefpane`? Otherwise this may simply be copied from inside the `.app` installer.

amatos commented:
Thank you for the advice.  I dug around some more with the app, and there's a app-installer-free download available that just pulls the prefpane directly.  I changed the cask to pull that edition, added the version tag and sha256 checksum, added an uninstall and zap stanzas.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,26 @@
+cask 'earthdeskprefpane' do
+  version '7.1'
```
Make this `vesion '710-27D8'`.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,26 @@
+  version '7.1'
+  sha256 'f5432c12e0dc23c5cf97debc807a1c05b88ac45253ace4ddc314652458e96064'
+
+  url 'http://download.xericdesign.com/earthdesk-710-27D8.zip'
```
And this `url "http://download.xericdesign.com/earthdesk-#{version}.zip"`.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,26 @@
+  homepage 'http://www.xericdesign.com/earthdesk.php'
+  license :commercial
+
+  auto_updates true
```
Can you add an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md)? If it `auto_updates`, it has one.

amatos commented:
Fixed the version tag, changed the download URL to use the version variable, and added the appcast stanza.

vitorgalvao commented on Casks/earthdeskprefpane.rb:
```diff
@@ -0,0 +1,28 @@
+  sha256 'f5432c12e0dc23c5cf97debc807a1c05b88ac45253ace4ddc314652458e96064'
+
+  url "http://download.xericdesign.com/earthdesk-#{version}.zip"
+  appcast 'http://www.xericdesign.com/sparkle/feeds/EarthDeskAppFeedV7.xml',
```
One final change. This can be `appcast "http://www.xericdesign.com/sparkle/feeds/EarthDeskAppFeedV#{version.major}.xml",`.

Thank you for the quick response time!

vitorgalvao commented:
Thank you for the quick response time!

Everything looks fine. Just waiting for Travis to finish, and we can merge.

amatos commented:
Updated the app cast to use version.major variable.

vitorgalvao commented:
> Updated the app cast to use version.major variable.

That was actually a mistake, hence why I retracted it. It worked with the previous version, but cannot work with the new one. The one you had was correct; using `#{version.major}` now will fail audit.

vitorgalvao commented:
As an additional note, having `#{version.major}` or any other interpolation in single quotes does not work: it does not expand and ruby takes it to be the literal string. Those cases require double quotes (like in `url`).

vitorgalvao commented:
Thank you for the contribution. It needed some fixes, so they were made in 9da80fd9bee0dac32956c39f4130711be7705cb6. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.
